src/Lean/Elab/Tactic/Simp.lean

```diff
@@ -180,6 +180,9 @@ def elabSimpArgs (stx : Syntax) (ctx : Simp.Context) (eraseLocal : Bool) (kind :
 
           match (← resolveSimpIdTheorem? term) with
           | .expr e  =>
+            if kind == .dsimp && !(← isRflProof e) && (← isProof e) then
+              -- TODO: Make this error message more informative, by stating which argument failed.
```

Comment:
This TODO seems unnecessary: you're already calling `throwErrorAt term` here, so the squiggle should come on the bad lemma.